108.Convert_Sorted_Array_to_BST.py

```diff
@@ -0,0 +1,19 @@
+# https://leetcode.com/problems/convert-sorted-array-to-binary-search
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+def sortedarraytobst(nums):
```

Comment:
when you pass nums[:c] to you function, it create a new array that spends a lot of memory
you can pass just indexes like that:
```
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        
        if not nums: return None
        
        def foo(left, right):
            if left > right:
                return None
            
            m = left + (right - left) // 2
            root = TreeNode(nums[m])
            root.left = foo(left, m-1)
            root.right = foo(m+1, right)
            return root
        
        root = foo(0, len(nums)-1)
        return root
```